base/src/main/java/org/kontinuity/catapult/base/identity/TokenIdentity.java

```diff
@@ -0,0 +1,21 @@
+package org.kontinuity.catapult.base.identity;
+
+/**
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+public class TokenIdentity implements Identity {
+    private final String token;
+
+    TokenIdentity(String token) {
+        this.token = token;
```

Comment:
IdentityFactory is the only place to create TokenIdentity objects. Maybe move the precondition checking here?